subsys/bluetooth/mesh/adv.c

```diff
@@ -159,22 +176,26 @@ struct net_buf *bt_mesh_adv_buf_get(k_timeout_t timeout)
 	return process_events(events, ARRAY_SIZE(events));
 }
 
-static struct net_buf *bt_mesh_adv_buf_relay_get(k_timeout_t timeout)
-{
-	return net_buf_get(&bt_mesh_relay_queue, timeout);
-}
-
 struct net_buf *bt_mesh_adv_buf_get_by_tag(uint8_t tag, k_timeout_t timeout)
 {
-	if (tag & BT_MESH_LOCAL_ADV) {
+	if (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE) && tag & BT_MESH_FRIEND_ADV) {
+		return net_buf_get(&bt_mesh_friend_queue, timeout);
+	}
+
+#if CONFIG_BT_MESH_RELAY_ADV_SETS
+	if (tag & BT_MESH_RELAY_ADV) {
+		return net_buf_get(&bt_mesh_relay_queue, timeout);
+	}
+#endif
+
+	if (tag & BT_MESH_LOCAL_ADV || tag & BT_MESH_FRIEND_ADV || tag & BT_MESH_RELAY_ADV ||
```

Comment:
On more small comment. Not sure if we need to check tag here. If we are not pooling friend neither relay message, then only locally created messages are left.

Reply:
> Feel free to disregard the comments regarding `IS_ENABLED` as `IS_ENABLED` should work for relay queue as well, but now both relay and friend implementations are aligned with each other.
> 
> Last question. Did you do sanity check of friendship with `CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE` option enabled? Or did you try bsim friendship tests with the new option?

I ran the entire mesh bsim suite with every cfg combination I could think of, and it seems fine if we disregard the packet collision in friendship/ msg_group.sh that we have discussed earlier.